larray/core.py

```diff
@@ -4685,6 +4685,78 @@ def rename(self, renames=None, to=None, inplace=False, **kwargs):
         else:
             return LArray(self.data, axes)
 
+    def reindex_axis(self, axis, labels, fill_value=np.nan, inplace=False):
+        """Change order of labels of an axis and/or add new ones.
+
+        Place NaN or given `fill_value` in locations having no value previously.
+
+        Parameters
+        ----------
+        axis : int, str or Axis
+            Axis along with to reindex.
+        labels : list of str
```

Comment:
the function should also accept an Axis argument, so that one can write: `arr.reindex_axis(x.a, other.a)`, which will probably be one of the main use case for this method.